src/scanpy/preprocessing/_simple.py

```diff
@@ -715,14 +762,28 @@ def regress_out(
             regres = regressors
         tasks.append(tuple((data_chunk, regres, variable_is_categorical)))
 
-    from joblib import Parallel, delayed
+    res = None
+    if not variable_is_categorical:
+        A = regres.to_numpy()
```

Comment:
I think the above code should be refactored as well to include these new methods in the condition (i.e. at https://github.com/scverse/scanpy/pull/3110/files#diff-f9e0bdcdfc04622c421f0a5788bbba6ee0303750580d2915caba3239d799322fR759).  We can just check `regressors.to_numpy()` for non-zero determinant before iterating through the chunk list.  We don't need to make the list at all, from what I can tell, if `not variable_is_categorical` and the determinant is non-zero.  This will also make things cleaner because we will need fewer checks.  We can then use a boolean to check whether we should move into the next clause instead of checking `res=None` or not.  

Does this make sense?